esphome/components/esp32/__init__.py

```diff
@@ -944,6 +955,14 @@ async def to_code(config):
             )
             add_idf_sdkconfig_option("CONFIG_IDF_EXPERIMENTAL_FEATURES", True)
 
+        # Set main loop stack size
+        if (
+            CONF_MAIN_LOOP_STACK_SIZE in advanced
+            and advanced.get(CONF_MAIN_LOOP_STACK_SIZE) > 8192
+        ):
```

Comment:
```suggestion
        if CONF_MAIN_LOOP_STACK_SIZE in advanced:
```

the schema will already have validated this since MIN_MAIN_LOOP_TASK_STACK_SIZE is set to 8192

Reply:
Thanks for the hint. I was a bit over enthusiastic on the failsafe that we never get below 8192. Even if someone changes it at once place (like where it has been documented) there was still this safeguard in place in Python and another one in preprocessor macros